geoapps_utils/driver/driver.py

```diff
@@ -114,7 +114,10 @@ def add_ui_json(self, entity: ObjectBase):
 
         :param entity: Object to add ui.json file to.
         """
-        entity.add_file(Path(self.params.input_file.path) / self.params.input_file.name)
+        assert self.params.input_file is not None
```

Comment:
we usually `assert` only in tests

```
if self.params.input_file is None:
    raise AttributeError("Params class needs an 'input_file'")
```

Reply:
I have added it so that mypy knows it not None
It is stating that it cannot be null if we reach that line. More like an invariant.

Also, assert are ignored when using the -O option at runtime. See https://docs.python.org/3/using/cmdline.html#cmdoption-o

Exception for sure makes sense when checking the input parameter at a high level. Not so sure here.